prime-router/docs/features/0004-managing-translation-schemas.md

```diff
@@ -0,0 +1,217 @@
+
+#### HL7 -> FHIR
+Long term, it would be great to add support to this library (it's open sourced) that would support reading files from different file-like storage solutions rather than just disk, but this would likely not be feasible to get done in a quick enough timeframe.  Instead, before instantiating the converter, the application will sync the azure storage to a spot on the disk the library will read from.
+
+#### Cache
+**This could potentially be deferred until it's been identified that the network requests to azure blob storage are the bottleneck**
+
+Since the pipeline will now require reading several files out of the blob store at multiple spots, it will likely make sense to cache the resolved schemas since they will change relatively infrequently.  This could initially be done with a simple in-memory cache using the schema URI as the cache key.
+
+#### Pros
+- Maintains the current file based model for the schemas
+- Schemas can be easily edited by hand via the Azure UI
+- Supports reading schemas from azure as well as from disk
+    - Enables leaving the common schemas on disk if that's preferred
+- Enables easy rollback
+    - Post MVP, a UI could be enabled to allow selecting what version to go back to
+- Supports reading files from azure or from disk
+
+#### Cons
+- Need to support to different URI schemes
+- Relies on pulling files out of azure which is slower than disk or the database
+- Requires more complicated error handling
+
+## Managing schemas
+The schemas break down into two categories:
+
+- schemas dedicated to a specific sender or receiver
+  - with one specifically being a "leaf" schema or a schema that is consumed by name in the pipeline and associated to a sender or receiver setting
+- schemas that are used across multiple senders or receivers across organizations
+    - in some cases, it might be preferable to keep some of these in the source to prevent accidentally breaking things
+
+and there will need to proper permissions placed such that schemas that are used across multiple senders and receivers can only be edited by prime admins
+
+### New APIs
+- Transform types `{transformType}` parameter
+    - HL7 -> FHIR: `hl7-fhir`
+    - FHIR -> FHIR: `fhir-fhir`
+    - FHIR -> HL7: `fhir-hl7`
+
+#### Sender/Receiver
+- Create a sender/receiver schema
+    - POST
+    - Auth: Prime admins, org admins
+    - `/v1/senders|receivers/{sender|receiverName}/schemas/{transformType}`
+    - Body
+        - `schemaName` string
+            - This is the path the schema should be placed at i.e. `metadata/hl7_mapping/common/patient.yml`
+        - `leafSchema` boolean
+            - If the field is set to true, set the `schemaName` property on the sender or receiver setting and indicates that this is a schema that will be consumed in the pipeline
+        - `schema` string
+        - `sampleInput` string
```

Comment:
For this initial cut, I was not intending on persisting the sample input/output.  It would be simply be a source of additional validation that gets run before we update/add the schema